### PR TITLE
rp2: make for different boards

### PR DESCRIPTION
**Pull request by jahr**
This change allows to build firmware for different RP2 based boards. So far only original Pico and Adafruit Feather RP2040 are added (I have no others to test). Board names are taken from pico-sdk/src/boards/include/boards/.

Usage:
Pico firmware can be build either using `make` as previously (it is default) or by `make BOARD=pico`. Feather is build by `make BOARD=adafruit_feather_rp2040`.
Only board name and flash drive size is set, pin definition is taken from the appropriate pico-sdk board definition.
Firmware is saved in the directory build-BOARD_NAME, e.g. build-pico, instead of build.

It seems that the pico-sdk has wrong flash size in Feather definition, it states 4MB, but it should be 8MB. Nonetheless I've set the flash size to 3MB and it works.

### CODE REVIEW

**Comment by dpgeorge** [comment]
See related #6987.

**Comment by dpgeorge** [comment]
Thanks for this, it's a needed addition.

To match the majority of the existing ports (stm32, esp32, samd, mimxrt) it would be good if the board names were capitalised, eg PICO, ADAFRUIT_FEATHER_RP2040.

It would also be good to be able to continue to use cmake on its own, ie without the `Makefile` wrapper (that's currently possible to do).  In that case the `CMakeLists.txt` file will need to contain logic to set the board name if it's not already set.  You can see how the esp32 port does this, eg:
```
# Set the board if it's not already set.
if(NOT MICROPY_BOARD)
    set(MICROPY_BOARD GENERIC)
endif()

# Set the board directory and check that it exists.
if(NOT MICROPY_BOARD_DIR)
    set(MICROPY_BOARD_DIR ${MICROPY_PORT_DIR}/boards/${MICROPY_BOARD})
endif()
if(NOT EXISTS ${MICROPY_BOARD_DIR}/mpconfigboard.cmake)
    message(FATAL_ERROR "Invalid MICROPY_BOARD specified: ${MICROPY_BOARD}")
endif()
```

File: ports/rp2/mpconfigport.h [hunk 1]
```diff
@@ -31,8 +31,12 @@
 #include "hardware/sync.h"
 #include "pico/binary_info.h"
 
+#include <mpconfigboard.h>
```
Comment (dpgeorge) [inline ports/rp2/mpconfigport.h:34]: Better to follow the other ports and use `"`'s here instead of `<>`.

**Comment by jahr** [comment]
Hello, I've made fixes based on the code review. Board names are now uppercase and defaults are added to `CMakeLists.txt`. Unfortunately I did not succeeded to build the board by running the cmake alone, so it is not tested. I added mpconfigboard.cmake to each board dir, they are empty now, perhaps PICO_DEFAULT_BOOT_STAGE2_FILE mentioned in #6987 may be set there, but I'm not sure if it is necessary.

**Comment by dpgeorge** [comment]
> Unfortunately I did not succeeded to build the board by running the cmake alone

It works for me!  I did:
```
$ cd ports/rp2
$ mkdir mybuild
$ cd mybuild
$ cmake ..
...
```
That builds the default PICO.  To specify another board do:
```
$ cmake .. -DMICROPY_BOARD=ADAFRUIT_FEATHER_RP2040
...
```

**Comment by dpgeorge** [comment]
Squashed and merged in 7ca686684ea5bffb45764bc3782f54133855a3c9

**Pull request closed without merging**